scripts/kind-setup.sh

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env bash
+
+# kind-setup.sh contains functions used to setup a KIND cluster and install 
+# a number of additional ACK service controllers (from their respective Helm 
+# repositories).
+
+SCRIPTS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+ROOT_DIR="$SCRIPTS_DIR/.."
+
+source "$SCRIPTS_DIR/lib/common.sh"
+source "$SCRIPTS_DIR/lib/config.sh"
+source "$SCRIPTS_DIR/lib/logging.sh"
+
+setup_kind_cluster() {
+    local __cluster_name=$1
+    local __controller_namespace=$2
+    local kubeconfig_path="$ROOT_DIR/build/clusters/$__cluster_name/kubeconfig"
+
+    info_msg "Creating cluster with name \"$__cluster_name\""
+    _create_kind_cluster $__cluster_name $kubeconfig_path
+
+    info_msg "Exporting KUBECONFIG=$kubeconfig_path"
+    export KUBECONFIG=$kubeconfig_path
+
+    _install_additional_controllers $__controller_namespace
+}
+
+_create_kind_cluster() {
+    local __cluster_name=$1
+    local __kubeconfig_path=$2
+
+    local config_file_name="$(get_cluster_configuration_file_name)"
+    local cluster_version="$(get_cluster_k8s_version)"
+
+    local config_file_path=$SCRIPTS_DIR/kind-configurations/$config_file_name
+    
+    info_msg "Using configuration \"$config_file_name\""
+    debug_msg "Using K8s version \"$cluster_version\""
+
+    for i in $(seq 0 3); do
```

Comment:
They come from the original script. I reduced it from 5 down to 3 retries. Sometimes KIND fails to bring up a cluster (I've had this happen many times). Usually the retries don't fix this, though, but I figured it was worth a shot in case it happens in Prow